articles/quickstart/webapp/php/01-login.md

```diff
@@ -11,159 +12,214 @@ topics:
-By default, the SDK stores user information in the PHP session and discards the access and ID Tokens.
+### Checking for a session
 
-To keep the tokens, to the SDK configuration, pass the following:
-* `'persist_access_token' => true`
-* `'persist_id_token' => true`
+The Auth0 PHP SDK has a convenient method for checking if our user has authenticated and returned their profile, `getCredentials()`. Let's install this on our index route to print the user profile if they're logged in or report that they need to login.
 
-To disable the session, pass `'store' => false` to the SDK configuration.
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
 
-Instead of using the PHP session to store information, you can use Laravel, Zend, Symfony or similar techniques. To do that, create a class that implements the get, set and delete methods and pass it to the SDK.
+function onIndexRoute() {
+  $session = $auth0->getCredentials();
 
-```php
-// index.php
-
-$laravelStore = new MyLaravelStore();
-$auth0 = new Auth0(array(
-    // ...
-    'store' => $laravelStore,
-    // ...
-));
+  if ($session === null) {
+    // The user isn't logged in.
+    echo 'Please <a href="/login">log in</a>.</p>';
+    return;
+  }
+
+  // The user is logged in.
+  print_r($session->user);
+
+  echo 'You can now <a href="/logout">log out</a>.</p>';
+}
+```
+
+We can access all the properties of our user's profile from the `user` property response, which is an array. So, for example, we can pull the user's nickname using `$session->user['nickname']`, or their email address from `$session->user['email']`. The structure is a [normalized user profile](https://auth0.com/docs/users/user-profile-structure), which you can learn more about [here](https://auth0.com/docs/users/normalized-user-profiles).
+
+It's important to note that the content of the user profile will vary depending on the social provider(s) you use, so you should never assume that a particular value will always be there within your application logic. Use PHP language constructs like `isset` or null coalescence to gracefully handle a value's presence, or lack thereof, for example:
+
+```PHP
+// ✋ We don't need to include this in our sample application, it's just an example.
+$name = $session->user['name'] ?? $session->user['nickname'] ?? $session->user['email'] ?? 'Unknown';
+```
+
+### Logging in
+
+Now let's create our /login route, which will use the Auth0 PHP SDK's `login()` method to setup the user session and return a customized URL to Auth0's Universal Login Page for this user to login.
+
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
+
+function onLoginRoute() {
+    // It's a good idea to reset user sessions each time they go to login to avoid "invalid state" errors, should they hit network issues or other problems that interrupt a previous login process:
+    $auth0->clear();
+
+    // Setup the user's session and generate a ULP URL:
```

Comment:
I suggest we avoid these acronyms, as we haven't introduced them above in the article and devs might get confused.

```suggestion
    // Setup the user's session and generate a Universal Login Page URL:
```